src/Reflection/ReflectionProperty.php

```diff
@@ -592,9 +671,123 @@ private function computeModifiers(PropertyNode $node): int
 
+        /** @phpstan-ignore return.type */
         return $modifiers;
     }
+
+    private function computeImmediateVirtual(PropertyNode $node): bool
+    {
+        if ($node->hooks === []) {
+            return false;
+        }
+
+        $setHook = null;
+
+        foreach ($node->hooks as $hook) {
+            if ($hook->name->name === 'set') {
+                $setHook = $hook;
+                break;
+            }
+        }
+
+        if ($setHook === null) {
+            return true;
+        }
+
+        $setHookBody = $setHook->getStmts();
+        assert($setHookBody !== null);
+
+        // Short syntax
+        if (count($setHookBody) === 1 && $setHookBody[0] instanceof Node\Stmt\Return_) {
+            return false;
+        }
+
+        $visitor   = new FindingVisitor(static fn (Node $node): bool => $node instanceof Node\Expr\Assign);
+        $traverser = new NodeTraverser($visitor);
+        $traverser->traverse($setHookBody);
+
+        foreach ($visitor->getFoundNodes() as $assigNode) {
+            assert($assigNode instanceof Node\Expr\Assign);
+            $variableToAssign = $assigNode->var;
+
+            if (
+                $variableToAssign instanceof Node\Expr\PropertyFetch
+                && $variableToAssign->var instanceof Node\Expr\Variable
+                && $variableToAssign->var->name === 'this'
+                && $variableToAssign->name instanceof Node\Identifier
+                && $variableToAssign->name->name === $this->name
+            ) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /** @return array{get?: ReflectionMethod, set?: ReflectionMethod} */
+    private function createImmediateHooks(PropertyNode $node): array
+    {
+        $hooks = [];
+
+        foreach ($node->hooks as $hook) {
+            $hookName = $hook->name->name;
+            assert($hookName === 'get' || $hookName === 'set');
+
+            $hooks[$hookName] = ReflectionMethod::createFromNode(
+                $this->reflector,
+                $hook,
+                $this->getDeclaringClass()->getLocatedSource(),
+                sprintf('$%s::%s', $this->name, $hookName),
+                null,
+                $this->getDeclaringClass(),
+                $this->getImplementingClass(),
+                $this->getDeclaringClass(),
+            );
+        }
+
+        return $hooks;
+    }
+
+    private function createCachedVirtual(): bool
+    {
+        if (! $this->immediateVirtual) {
+            return false;
+        }
+
+        return $this->getParentProperty()?->isVirtual() ?? true;
+    }
+
+    /** @return array{get?: ReflectionMethod, set?: ReflectionMethod} */
+    private function createCachedHooks(): array
+    {
+        $hooks = $this->immediateHooks;
+
+        if (isset($hooks['get'], $hooks['set'])) {
+            return $hooks;
+        }
```

Comment:
Perhaps better to check if `$hooks === []`?

Reply:
The only situation when I don't need to check parent class is when I have both hooks.

I've added a comment to clarify this is just optimization.